### PR TITLE
[GLUTEN-6923][CH] `total_bytes_written` is not updated in celeborn partition writers

### PR DESCRIPTION
**Pull request by lgbo-ustc**
## What changes were proposed in this pull request?

(Please fill in changes proposed in this fix)

Fixes: #6923

AQE has dependency on shuffle written bytes to generate a proper plan. 

## How was this patch tested?

(Please explain how this patch was tested. E.g. unit tests, integration tests, manual tests)

manual tests

(If this patch involves UI changes, please attach a screenshot; otherwise, remove this)

### CODE REVIEW

**Comment by baibaichen** [comment]
能加 UT 吗？

**Comment by lgbo-ustc** [comment]
> 能加 UT 吗？

这要用到celeborn，现在没有集成测试，加不了吧

**Review by liuneng1994 — APPROVED** [review]
LGTM

**Comment by baibaichen** [comment]
Run Gluten Clickhouse CI

File: backends-clickhouse/src/main/scala/org/apache/gluten/backendsapi/clickhouse/CHSparkPlanExecApi.scala [hunk 1]
```diff
@@ -539,6 +539,15 @@ class CHSparkPlanExecApi extends SparkPlanExecApi {
       CHExecUtil.buildSideRDD(dataSize, newChild).collect
 
     val batches = countsAndBytes.map(_._2)
+    val totalBatchesBytes = batches.map(_.length).sum
+    if (
+      totalBatchesBytes < 0 || totalBatchesBytes.toLong > CHBackendSettings.getBroadcastThreshold
+    ) {
+      throw new SparkException(
```
Comment (exmy) [inline backends-clickhouse/src/main/scala/org/apache/gluten/backendsapi/clickhouse/CHSparkPlanExecApi.scala:546]: use `GlutenException`?

**Comment by lgbo-ustc** [comment]
Run Gluten Clickhouse CI

**Comment by lgbo-ustc** [comment]
Run Gluten Clickhouse CI

**Comment by lgbo-ustc** [comment]
Run Gluten Clickhouse CI

**Comment by lgbo-ustc** [comment]
Run Gluten Clickhouse CI

**Comment by lgbo-ustc** [comment]
Run Gluten Clickhouse CI

**Comment by lgbo-ustc** [comment]
Run Gluten Clickhouse CI

**Comment by lgbo-ustc** [comment]
Run Gluten Clickhouse C

File: backends-clickhouse/src/main/scala/org/apache/gluten/backendsapi/clickhouse/CHSparkPlanExecApi.scala [hunk 2]
```diff
@@ -539,10 +541,16 @@ class CHSparkPlanExecApi extends SparkPlanExecApi {
       CHExecUtil.buildSideRDD(dataSize, newChild).collect
 
     val batches = countsAndBytes.map(_._2)
+    val totalBatchesSize = batches.map(_.length).sum
     val rawSize = dataSize.value
-    if (rawSize >= BroadcastExchangeExec.MAX_BROADCAST_TABLE_BYTES) {
-      throw new SparkException(
-        s"Cannot broadcast the table that is larger than 8GB: ${rawSize >> 30} GB")
+    if (rawSize >= BroadcastExchangeExec.MAX_BROADCAST_TABLE_BYTES ||) {
```
Comment (liuneng1994) [inline backends-clickhouse/src/main/scala/org/apache/gluten/backendsapi/clickhouse/CHSparkPlanExecApi.scala:546]: compile error

**Comment by lgbo-ustc** [comment]
Run Gluten Clickhouse C

**Pull request merged**